components/server/src/express/ws-handler.ts

```diff
@@ -107,4 +114,21 @@ export class WsExpressHandler {
         }
         return pathname === matcher;
     }
+
+    /**
+     * Stop accepting new connections, returns when all connections are closed.
+     */
+    dispose(): Promise<void> {
+        return new Promise((resolve, reject) => {
```

Comment:
dispose is never waiting for a promise so retuning this doesn't have an effect. Do we want to introduce an AsyncDisposable?

Reply:
or we do it explicitly, i.e. call WsHandler.close or something like that